crypto/params_dup.c

```diff
@@ -234,3 +235,12 @@ void OSSL_PARAM_free(OSSL_PARAM *params)
         OPENSSL_free(params);
     }
 }
+
+size_t ossl_param_nelem(const OSSL_PARAM *plist)
+{
+    size_t n;
+
+    for (n = 0; !ossl_param_is_empty(plist); plist++)
```

Comment:
This is probably just better using plist->key == NULL

Reply:
It an inline function, so shouldn't cost anything.  Doing it this way also covers the `plist == NULL` case without extra lines of code.